tests/rptest/scale_tests/upgrade_availability_test.py

```diff
@@ -0,0 +1,243 @@
+
+        # The simple producer uses Kafka 2.8.0
+        self.producer = KafkaSimpleProducer(self.test_context,
+                                            self.redpanda,
+                                            self.topic,
+                                            runtime=runtime)
+        self.producer.start()
+
+    def move_from_src_to_dest(self, src_id: int, dest_id: int):
+        self.logger.info(f'Moving replica from node {src_id} to {dest_id}')
+        self.assignments = self._move_replica(self.topic, 0, src_id, dest_id)
+
+    def run_workload_and_get_replica_sets(self):
+        # Start workload
+        self.start_workload(runtime=180)
+
+        orig_assignments = self._get_assignments(self.admin, self.topic, 0)
+        nodes_in_replica_set = [a["node_id"] for a in orig_assignments]
+        brokers = self.admin.get_brokers()
+
+        # Wait for some load
+        wait_until(lambda: self.producer.record_count > 50000,
+                   timeout_sec=60,
+                   err_msg='Failed to write enough data')
+
+        nodes_not_in_replica_set = [
+            b["node_id"] for b in brokers
+            if b["node_id"] not in nodes_in_replica_set
+        ]
+
+        return nodes_in_replica_set, nodes_not_in_replica_set
+
+    def confirm_in_progress_partition_movement(self, hosts: list[str]):
+        def check_status_on_all_replicas():
+            replicas_status = []
+
+            for host in hosts:
+                res = self.admin._get_configuration(host,
+                                                    namespace='kafka',
+                                                    topic=self.topic,
+                                                    partition=0)
+
+                if 'status' not in res:
+                    return False
+                else:
+                    status = res['status']
+                    self.logger.debug(f'Movement status: {status}')
+                    replicas_status.append(status == 'in_progress')
+
+            return all(replicas_status)
+
+        wait_until(check_status_on_all_replicas, timeout_sec=120)
+
+    def restart_node_and_wait_partition_movement(self, node_id: int):
+        self.logger.info(f'Restarting node {node_id} with version 22.1.3')
+        node_to_upgrade = self.redpanda.get_node(node_id)
+        self.restart_node_w_version(node_to_upgrade, '22.1.3-1')
+
+        # Make sure partition movement is still in progress
+        self.confirm_in_progress_partition_movement(hosts=self.rp_hostnames)
+
+        # Make sure the producer is still writing
+        records_written_after_upgrade = self.producer.record_count
+        wait_until(
+            lambda: self.producer.record_count > records_written_after_upgrade,
+            timeout_sec=60,
+            err_msg='Producer stopped writing')
+
+        # Wait for partition movement to complete
+        self.admin.wait_stable_configuration(topic=self.topic,
```

Comment:
stable configuration means that all nodes returns the same data but it doesn't mean that status is `done` we need to explicitly check for it

Reply:
done.